[shopsys] added change and remove methods to js register class (#1676)
shopsys/shopsys: docs/frontend/npm-and-webpack.md

```

This principle is called [Monkey Patching](https://www.sitepoint.com/pragmatic-monkey-patching/).

### I want to override class from @shopsys/framework common package

> You can use ES6 syntax for override class. You certainly know key word `extend`. You can use it in es world now.

Reviewer: ```suggestion
You can use ES6 syntax to override class.
You certainly know the key word `extend`.
You can use it in the ES world now.
```

Author: 7872665e959a169f7ab08abe7e64147a91d51fac

```

This principle is called [Monkey Patching](https://www.sitepoint.com/pragmatic-monkey-patching/).

### I want to override class from @shopsys/framework common package

You can use ES6 syntax to override class.
You certainly know key word `extend`.
You can use it in es world now.

```js
import CategoryTreeSorting from 'framework/admin/components/categoryTree.sorting';
import Register from 'framework/common/utils/register';

class MyCategoryTreeSorting extends CategoryTreeSorting {
    constructor () {
        super();
        console.log('patched constructor');
    }

    init () {
        new MyCategoryTreeSorting();
    }
}
```

> If you override framework's common javascript class, you will have to change registered callback with original class init.

Reviewer: ```suggestion
If you override framework's javascript class, you will have to change registered callback of the original class init to your implementation.
```
